packages/core-bridge/src/runtime.rs

```diff
@@ -386,7 +398,18 @@ pub fn runtime_new(mut cx: FunctionContext) -> JsResult<BoxedRuntime> {
     let channel = Arc::new(cx.channel());
     let (sender, mut receiver) = unbounded_channel::<RuntimeRequest>();
 
-    std::thread::spawn(move || start_bridge_loop(telemetry_options, channel, &mut receiver));
+    // FIXME: This is a temporary fix to get sync notifications of errors while initializing the runtime.
+    //        The proper fix would be to avoid spawning a new thread here, so that start_bridge_loop
+    //        can simply yeild back a Result. But early attempts to do just that caused panics
+    //        on runtime shutdown, so let's use this hack until we can dig deeper.
+    let (result_sender, mut result_receiver) = unbounded_channel::<Result<(), String>>();
+
+    std::thread::spawn(move || start_bridge_loop(telemetry_options, channel, &mut receiver, result_sender));
+
+    if let Some(Err(e)) = result_receiver.blocking_recv() {
+        Err(cx.throw_error::<_, String>(e).unwrap_err())?;
+    }
+    result_receiver.close();
```

Comment:
You don't need to explicitly close this.

Alternatively, you could use a oneshot channel